api/bucketsd/service.go

```diff
@@ -970,16 +973,175 @@ func (s *Service) SetPath(ctx context.Context, req *pb.SetPathRequest) (res *pb.
 	}
 	dbToken, _ := thread.TokenFromContext(ctx)
 
-	destPath := cleanPath(req.Path)
-	bootCid, err := cid.Decode(req.Cid)
+	if isEqualPath(req.Path, req.Destination) {
+		return &pb.MovePathResponse{}, nil
+	}
+
+	cleanPth, err := parsePath(req.Path)
 	if err != nil {
-		return nil, fmt.Errorf("invalid remote cid: %v", err)
+		return nil, err
+	}
+	cleanDst, err := parsePath(req.Destination)
+	if err != nil {
+		return nil, err
 	}
 
 	lck := s.Semaphores.Get(buckLock(req.Key))
 	lck.Acquire()
 	defer lck.Release()
 
+	buck, pth, err := s.getBucketPath(ctx, dbID, req.Key, cleanPth, dbToken)
+	if err != nil {
+		return nil, err
+	}
+	n, err := s.getNodeAtPath(ctx, pth, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = s.setPath(ctx, &pb.SetPathRequest{
+		Key:  req.Key,
+		Path: cleanDst,
+		Cid:  n.Cid().String(),
+	}, dbID, dbToken, cleanDst)
+	if err != nil {
+		return nil, fmt.Errorf("error setting path: %v", err)
+	}
+	if buck.IsPrivate() {
+		meta, _, ok := buck.GetMetadataForPath(cleanPth, false)
+		if !ok {
+			return nil, fmt.Errorf("error getting metadata")
+		}
+		buck.SetMetadataAtPath(cleanDst, meta)
+		if err = s.Buckets.Verify(ctx, dbID, buck, tdb.WithToken(dbToken)); err != nil {
+			return nil, fmt.Errorf("verifying bucket update: %v", err)
+		}
+	}
```

Comment:
I think you can skip this since `setPath` is already doing the verification step.